Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1977,6 +1979,25 @@ PhysicalParticleContainer::getIonizationFunc (const WarpXParIter& pti,
                                 ion_atomic_number);
 }
 
+void PhysicalParticleContainer::resample (Resampling& resampler, const int timestep)
+{
+    const amrex::Real global_numparts = TotalNumberOfParticles();
+
+    if (resampler.triggered(timestep, global_numparts))
+    {
+        auto & warpx = WarpX::GetInstance();
+        for (int lev = 0; lev <= warpx.maxLevel(); lev++)
+        {
+            for (WarpXParIter pti(*this, lev); pti.isValid(); ++pti)
+            {
+                resampler.run(pti);
```

Comment:
If you implement `run` as `operator()(...)` then this line would be:
```suggestion
                resampler(pti);
```